src/components/Articles/Articles.jsx

```diff
@@ -15,31 +18,54 @@ const Articles = () => {
     });
   }, []);
 
+  const handleArticleClick = (article_id) => {
+    setIsLoading(true);
+    getArticleById(article_id).then((articleFromApi) => {
+      setArticle(articleFromApi);
+      setIsLoading(false);
+      navigate(`/article/${article_id}`);
+    });
+  };
+
   return (
     <section className="articles-section">
-      {isLoading && <LoadingSpinner />}
-      <ol>
-        {articles[0] && (
-          <li key={articles[0].article_id} className="main-article">
-            <img src={articles[0].article_img_url} />
-            <h2>{articles[0].title}</h2>
-            <h4>Created by {articles[0].author} on {articles[0].created_at.slice(0, 10)}</h4>
-            <p>Topic: {articles[0].topic}</p>
+      {isLoading ? (
+        <LoadingSpinner />
+      ) : (
+        <ol>
```

Comment:
I like the approach here, but it is very repetitive and not DRY, you have an ArticleCard, consider whether it could be reusable here, or whether to create an ArticleListItem compent to keep this component more concise.